### ISSUE TITLE
Extend `pybamm.ParameterValues.create_from_bpx()` to automate initial concentrations for blended electrodes

### ISSUE THREAD

**Issue by ejfdickinson** [pybamm-team/PyBaMM#4338]
### Description

Calling `pybamm.ParameterValues.create_from_bpx()` with a BPX file containing a blended electrode returns a warning as follows:

`Initial concentrations cannot be set using stoichiometry limits for blend electrodes. Please set the initial concentrations manually.`

The method could automate this calculation by solving the equations specified in the BPX standard.

**Comment by rtimms** [comment]
Tagging https://github.com/pybamm-team/PyBaMM/issues/2682 as related. Currently the electrode state of health models in pybamm don't support blended electrodes.

**Comment by ejfdickinson** [comment]
@rtimms Off-topic I know, but for this reason I find it odd that `calc_esoh=True` is the default in `pybamm.Simulation.solve()`.

**Comment by rtimms** [comment]
Yes, this is simply because `calc_esoh` predates blended electrodes. I’m open to changing this.

**Comment by valentinsulzer** [comment]
The proper fix, which we are planning to do, is to do "just in time" calculations of esoh values only when they are requested
